src/app/lists/EditModal.tsx

```diff
@@ -55,25 +57,70 @@ const listQuantityNoteAndLabels: [keyof Schema["lists"], keyof Schema["lists"],
 const EditModal: React.FC<Props> = ({ data, onClose }) => {
     const [toSubmit, setToSubmit] = useState<Partial<Schema["lists"]>>(data ?? {});
 
-    const [errorMsg, setErrorMsg] = useState<string | null>(null);
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
     const setKey = (event: React.ChangeEvent<HTMLInputElement>, key: string): void => {
         const newValue = event.target.value;
         setToSubmit({ ...toSubmit, [key]: newValue });
     };
 
     const onSubmit = async (): Promise<void> => {
-        const table = supabase.from("lists");
-        const { error } =
-            data === null
-                ? await table.insert(toSubmit)
-                : await table.update(toSubmit).eq("primary_key", toSubmit.primary_key);
-
-        if (error) {
-            setErrorMsg(error.message);
+        if (data === null) {
+            const { data: returnedListData, error: insertListItemError } = await supabase
+                .from("lists")
+                .insert(toSubmit)
+                .select()
+                .single();
+
+            const auditLog = {
+                content: { itemDetails: toSubmit },
+                action: "add a list item",
+            } as const satisfies Partial<AuditLog>;
+
+            if (insertListItemError) {
+                const logId = await logErrorReturnLogId("failed to insert list item", {
+                    error: insertListItemError,
+                });
+                await sendAuditLog({ ...auditLog, wasSuccess: false, logId });
+                setErrorMessage(`Error: failedToInsertListItem. Log ID: ${logId}`);
+                return;
+            }
+
+            await sendAuditLog({
```

Comment:
Since the subsequent operations don't need to wait for this to finish...
```suggestion
            void sendAuditLog({
```